pkg/nimble/netif/nimble_netif.c

```diff
@@ -615,10 +620,10 @@ int nimble_netif_close(int handle)
     return NIMBLE_NETIF_OK;
 }
 
-int nimble_netif_accept(const uint8_t *ad, size_t ad_len,
-                        const struct ble_gap_adv_params *adv_params)
+static int _accept(const uint8_t *ad, size_t ad_len, const ble_addr_t *addr,
+                   uint32_t timeout,
+                   const struct ble_gap_adv_params *adv_params)
 {
-    assert(ad);
```

Comment:
Why can this now be NULL?

Reply:
Because `nimble_netif_accept_direct()` does not use a payload, and thus this shard backend function does accept empty advertising data. But I should move this check up into `nimble_netif_accept()` -> will do